spec/dds_md5_reporter_spec.rb

```diff
@@ -90,14 +122,66 @@
   end
 
   describe 'interface' do
-    subject {
+    let(:reporter) {
       DdsMd5Reporter.new(
-        upload_id: upload_id,
+        file_version_id: file_version_id,
         user_key: user_key,
         agent_key: agent_key,
         dds_api_url: dds_api_url
       )
     }
+    subject { reporter }
+
+    describe '#raise_dds_api_exception' do
+      it { is_expected.to respond_to(:raise_dds_api_exception) }
+
+      describe 'behavior' do
+        let(:resp) { double() }
+        let(:expected_preamble) { 'error preamble' }
+
+        before do
+          expect(resp).to receive(:body)
+            .and_return(expected_body)
+        end
+
+        context 'with dds error json response' do
+          let(:dds_error_report) {
+            {
+              "error" => "the error",
+              "reason" => "the reason",
+              "suggestion" => "the suggestion"
+            }
+          }
+          let(:expected_body) {
+            dds_error_report.to_json
```

Comment:
Does `to_json` create a single or multi-line json string? I *think* that DDS serializes to multi-line json, but I'm not sure about that.

Reply:
I built the expectation off of actual testing. The main expectation is that:
- resp.body.match(/error.*reason.*suggestion/) returns true
- resp.parsed_response deserializes the body to a hash with stringified keys
I verified both on an actual responses from calling the dds api

Reply:
Ok, just wanted to be sure, since ruby regex is picky about single vs multi-line strings.